pyroSAR/drivers.py

```diff
@@ -1910,34 +1910,37 @@ class TSX(ID):
     """
     
     def __init__(self, scene):
-        self.scene = os.path.realpath(scene)
-        
-        self.pattern = r'^(?P<sat>T[DS]X1)_SAR__' \
-                       r'(?P<prod>SSC|MGD|GEC|EEC)_' \
-                       r'(?P<var>____|SE__|RE__|MON1|MON2|BTX1|BRX2)_' \
-                       r'(?P<mode>SM|SL|HS|HS300|ST|SC)_' \
-                       r'(?P<pols>[SDTQ])_' \
-                       r'(?:SRA|DRA)_' \
-                       r'(?P<start>[0-9]{8}T[0-9]{6})_' \
-                       r'(?P<stop>[0-9]{8}T[0-9]{6})(?:\.xml|)$'
-        
-        self.pattern_ds = r'^IMAGE_(?P<pol>HH|HV|VH|VV)_(?:SRA|FWD|AFT)_(?P<beam>[^\.]+)\.(cos|tif)$'
-        self.examine(include_folders=False)
-        
-        if not re.match(re.compile(self.pattern), os.path.basename(self.file)):
-            raise RuntimeError('folder does not match TSX scene naming convention')
-        
-        self.meta = self.scanMetadata()
-        self.meta['projection'] = crsConvert(4326, 'wkt')
-        
-        super(TSX, self).__init__(self.meta)
+        if isinstance(scene, str):
+            self.scene = os.path.realpath(scene)
+        
+            self.pattern = r'^(?P<sat>T[DS]X1)_SAR__' \
+                        r'(?P<prod>SSC|MGD|GEC|EEC)_' \
+                        r'(?P<var>____|SE__|RE__|MON1|MON2|BTX1|BRX2)_' \
+                        r'(?P<mode>SM|SL|HS|HS300|ST|SC)_' \
+                        r'(?P<pols>[SDTQ])_' \
+                        r'(?:SRA|DRA)_' \
+                        r'(?P<start>[0-9]{8}T[0-9]{6})_' \
+                        r'(?P<stop>[0-9]{8}T[0-9]{6})(?:\.xml|)$'
+            
+            self.pattern_ds = r'^IMAGE_(?P<pol>HH|HV|VH|VV)_(?:SRA|FWD|AFT)_(?P<beam>[^\.]+)\.(cos|tif)$'
+            self.examine(include_folders=False)
+            
+            if not re.match(re.compile(self.pattern), os.path.basename(self.file)):
+                raise RuntimeError('folder does not match TSX scene naming convention')
+            
+            self.meta = self.scanMetadata()
+            self.meta['projection'] = crsConvert(4326, 'wkt')
+            
+            super(TSX, self).__init__(self.meta)
```

Comment:
the `super` call needs to be outdented so that it is not only called when `isinstance(scene, str)` but in all cases (e.g. when `scene` is a dictionary). Otherwise the class `TDM` cannot make use of the methods implemented by class `ID`.
```diff
-            super(TSX, self).__init__(self.meta)
+        super(TSX, self).__init__(self.meta)
```

Reply:
Only this one change and then it should be good to go.